remoteworkitem/trackerquery_repository.go

```diff
@@ -28,29 +28,29 @@ func NewTrackerQueryRepository(db *gorm.DB) *GormTrackerQueryRepository {
 	return &GormTrackerQueryRepository{db}
 }
 
+// TrackerQueryRepository encapsulate storage & retrieval of tracker queries
+type TrackerQueryRepository interface {
+	repository.Exister
+	Create(ctx context.Context, query string, schedule string, tracker uuid.UUID, spaceID uuid.UUID) (*app.TrackerQuery, error)
+	Save(ctx context.Context, tq app.TrackerQuery) (*app.TrackerQuery, error)
+	Load(ctx context.Context, ID string) (*app.TrackerQuery, error)
+	Delete(ctx context.Context, ID string) error
+	List(ctx context.Context) ([]*app.TrackerQuery, error)
+}
+
 // Create creates a new tracker query in the repository
 // returns BadParameterError, ConversionError or InternalError
-func (r *GormTrackerQueryRepository) Create(ctx context.Context, query string, schedule string, tracker string, spaceID uuid.UUID) (*app.TrackerQuery, error) {
-	tid, err := strconv.ParseUint(tracker, 10, 64)
-	if err != nil || tid == 0 {
-		// treating this as a not found error: the fact that we're using number internal is implementation detail
-		return nil, NotFoundError{"tracker", tracker}
-	}
-
-	log.Info(ctx, map[string]interface{}{
-		"tracker_id": tid,
-	}, "Tracker ID to be created")
-
+func (r *GormTrackerQueryRepository) Create(ctx context.Context, query string, schedule string, tracker uuid.UUID, spaceID uuid.UUID) (*app.TrackerQuery, error) {
```

Comment:
can you please rename `tracker` input parameter to `trackerID`, it confuses a bit.

Reply:
@pranavgore09 yes i will.

Reply:
fixed it here -> 
2824ae1